phpcs.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<ruleset name="PHP Compatibility Checker PHPCS Rules">
+	<description>A custom set of PHPCS rules for PHP Compatibility Checker.</description>
+
+	<exclude-pattern>*/tests/*</exclude-pattern>
+	<exclude-pattern>*/vendor/*</exclude-pattern>
+
+	<rule ref="WordPress"/>
+
+	<rule ref="WordPress.WP.I18n">
+		<properties>
+			<property name="text_domain" type="string" value="php-compatibility-checker"/>
+		</properties>
+	</rule>
+
+	<config name="testVersion" value="5.2-99.0"/>
+	<rule ref="PHPCompatibility"/>
```

Comment:
Please use a ruleset which takes backfills provided from within WP into account.... (should be done for the ruleset used to test plugins & themes as well actually...)

For an example, see: https://github.com/wimg/PHPCompatibility/issues/514#issuecomment-340369984 based on PHPCompatibility 8.1.0 which is expected within the week.

Reply:
Are you saying we could use exactly the same ruleset for PHPCompatibility that you show in that example?

Reply:
That would depend on the version of PHPCompatibility being used.
If `master` / `8.1.0` (once released), yes. For older versions, you'd have to remove some of the excludes as they refer to sniffs which didn't exist yet and PHPCS will complain about that.

As a side-note: I'm not claiming completeness with the example ruleset, there may be more things which are being back-filled by WP.
If you find any more, it would be appreciated to keep me informed - issue https://github.com/wimg/PHPCompatibility/issues/530 would probably be the most appropriate place for now.

Also: the example ruleset is based on WP 4.9, so may not be valid for older WP versions. You'd need to check which WP version is supported by a plugin and check which of the backfills was added in which WP version to be a 100% accurate ruleset-wise.

Reply:
I think this is a great idea but shouldn't block this PR.  I'd like to proceed and I'll open a new issue for this after the merge.